### PR TITLE
QueryInspector: add common way to show the raw query

### PR DESCRIPTION
**Pull request by ryantxu**
The query inspector currently just shows the raw request/response to the server.  Another common pattern is for data sources to add the escaped query text (sql, nrql, flux, etc) to the result metadata and then either let people find it in the HTTP response, or show it in a custom UI component.

This PR makes a standard way to do that for all datasources:
image

Verify it works as expected in:
- [x] PostgreSQL
- [x] MySQL
- [ ] MSSQL

### CODE REVIEW

File: public/app/features/dashboard/components/Inspector/QueryInspector.tsx [hunk 1]
```diff
@@ -47,6 +56,30 @@ export class QueryInspector extends PureComponent<Props, State> {
...
+    if (this.props.data !== oldProps.data) {
+      this.updateQueryList();
+    }
+  }
+
+  updateQueryList() {
+    const { data } = this.props;
+    const executedQueries: ExecutedQueryInfo[] = [];
+    if (data?.length) {
+      data.forEach((frame, idx) => {
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/Inspector/QueryInspector.tsx:72]: Nice, didn't know we had `forEach` on dataFrame 🎉 ⭐

File: public/app/features/dashboard/components/Inspector/QueryInspector.tsx [hunk 2]
```diff
@@ -202,6 +251,7 @@ export class QueryInspector extends PureComponent<Props, State> {
             new query. Hit refresh button below to trigger a new query.
           </p>
         </div>
+        {executedQueries.length && this.renderExecutedQueries(executedQueries)}
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/Inspector/QueryInspector.tsx:254]: nit: maybe move the length check to `this.renderExecutedQueries`?
```suggestion
        {this.renderExecutedQueries(executedQueries)}
```

File: public/app/features/dashboard/components/Inspector/QueryInspector.tsx [hunk 3]
```diff
@@ -182,8 +215,24 @@ export class QueryInspector extends PureComponent<Props, State> {
...
+          return (
+            <div key={info.query}>
+              {showName && <h3>{info.name}</h3>}
+              <pre>{info.query}</pre>
+            </div>
+          );
+        })}
+      </div>
+    );
+  }
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/Inspector/QueryInspector.tsx:232]: nit: moving the length check into this function simplifies the code a bit.
```suggestion
  renderExecutedQueries(executedQueries: ExecutedQueryInfo[]) {
    if(!executedQueries.length) {
      return null;
    }
    return (
      <div>
        {executedQueries.map(info => {
          return (
            <div key={info.query}>
              <h3>{info.name}</h3>
              <pre>{info.query}</pre>
            </div>
          );
        })}
      </div>
    );
  }
```

**Review by hugohaggmark — COMMENTED** [review]
Looks good, I just left some minor suggestions but I also get a bit confused when I try this it out.

1. I think we're missing the `pointer` cursor for `MSSQL` when you hover over the `Generated SQL >`
2. The name in QueryInspector differs from the RefId in QueryEditor so it makes my eyes look for something that doesn't exist. 
image
image
3. At some point (I couldn't find out how to repro again though) I got this strange behavior as the movie below:
generated-sql-confused

File: public/app/features/dashboard/components/Inspector/QueryInspector.tsx (hunk 1 above)
Comment (ryantxu) [reply public/app/features/dashboard/components/Inspector/QueryInspector.tsx:72]: data is an array, so don't be too impressed ;)

do check [vectorator](https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/vector/FunctionalVector.ts#L45)... that will let you do all the functional things you want with a vector

**Comment by ryantxu** [comment]
thanks @hugohaggmark -- good points.  I updated it so it shows the refId instead:  
image

File: public/app/features/dashboard/components/Inspector/QueryInspector.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/components/Inspector/QueryInspector.tsx:79]: thanks, will check it out!

**Review by hugohaggmark — APPROVED** [review]
Impressive work on this, much clearer now!

**Review by torkelo — APPROVED** [review]
Looks good! I like that it removes code from every SQL query editor to make them shared in inspector :)

**Pull request merged**